pkg/agent/nodeportlocal/portcache/port_table.go

```diff
@@ -64,94 +93,152 @@ func NewPortTable(start, end int) (*PortTable, error) {
 func (pt *PortTable) CleanupAllEntries() {
 	pt.tableLock.Lock()
 	defer pt.tableLock.Unlock()
-	pt.Table = make(map[int]NodePortData)
+	pt.NodePortTable = make(map[int]*NodePortData)
+	pt.PodEndpointTable = make(map[string]*NodePortData)
 }
 
-func (pt *PortTable) GetEntry(nodeport int) *NodePortData {
+func (pt *PortTable) GetEntry(ip string, port int, protocol string) *NodePortData {
 	pt.tableLock.RLock()
 	defer pt.tableLock.RUnlock()
-	data, _ := pt.Table[nodeport]
-	return &data
+	return pt.getEntryByPodIPPortProtocol(ip, port, protocol)
 }
```

Comment:
this function should be "private" (name starts with lower case letter)
it doesn't seem like a very useful function to have (I think I pointed that out earlier): you only call it once in `RuleExists`?

Reply:
it is getting called from `npl_controller.go` as well.